plugins/redfish/fu-redfish-backend.c

```diff
@@ -224,6 +225,63 @@ fu_redfish_backend_check_wildcard_targets(FuRedfishBackend *self)
 	}
 }
 
+static void
+fu_redfish_backend_set_session_key(FuRedfishBackend *self, const gchar *session_key)
+{
+	g_free(self->session_key);
+	self->session_key = g_strdup(session_key);
+}
+
+static size_t
+fu_redfish_backend_headers_callback(char *ptr, size_t size, size_t nmemb, void *userdata)
+{
+	gchar *session_key = NULL;
+	FuRedfishBackend *self = (FuRedfishBackend *)userdata;
```

Comment:
can you do `FU_REDFISH_BACKEND()` here rather than `(FuRedfishBackend *)` pls -- then we get some runtime type checking.